### PR TITLE
Added Background, StepArguments, and a Runner

### PR DESCRIPTION
**Pull request by timoteipalade**
This might be a bit much, but here is what I added:
I needed a way to run feature files and match the steps to step implementations, like XCTest-Gherkin does. 

So I extended the parser to support Background and Step Arguments (Datatable, Docstring).
Then I added a runner on top that parses a feature file, maps the steps to their implementations and runs them. 

The syntax used for the step definitions is a subset of the syntax of Cucumber Expressions. The supported keywords are: {int},  {float}, and {word}. I might add support for {} and {string} in the future.

### CODE REVIEW

**Comment by timoteipalade** [comment]
@iainsmith I would appreciate your input on this. Thanks!

**Comment by timoteipalade** [comment]
TODO: Fix the travis errors.

**Comment by iainsmith** [comment]
Hey @timoteipalade, apologies for missing this one. I’ll try and take a look sometime this week.

**Comment by timoteipalade** [comment]
@iainsmith no worries. I closed the PR for 4 reasons: 
1. I use some API that is not available on Linux. (Did not look into this in detail)
2. I did not have time to add compatibility for Swift 4.0 and 4.1. (Not much work here though)
3. It seemed like way too much code in a single PR. I should have branched before pushing more code.
4. I was not particularly careful. I have some mistakes, some leftover comments and unnecessary files.

You are still welcome to look over it. The first 2 commits might be of interest to you.

File: Sources/Gherkin/Transform.swift [hunk 1]
```diff
@@ -20,6 +21,24 @@ func _transform(label: GherkinLabel, values: [Any]) -> Any? {
     case .step:
         let name = StepName(rawValue: (values[0] as! String).lowercased())!
         let text = values[1] as! String
+        
+        //handle table
+        if values.count > 2, let keys = values[2] as? [String] {
```
Comment (timoteipalade) [inline Sources/Gherkin/Transform.swift:27]: This if statement is not necessary.

File: Tests/GherkinTests/GherkinTests.swift [hunk 2]
```diff
@@ -292,6 +292,52 @@ final class SwiftGherkinTests: XCTestCase {
 
         XCTAssertNotNil(feature)
     }
+    
+    func testSimpleFeatureWithDataTable() throws {
+        //TODO
```
Comment (timoteipalade) [inline Tests/GherkinTests/GherkinTests.swift:297]: Should erase this comment.

**Pull request closed without merging**